File: src/Running_Sum_of_1d_Array/solution.ts
/**
 * Returns a running sum of numbers (Imperative Solution)
 * @param {number[]} nums
 * @returns {number[]} - Returns an array of numbers
 */

const runningSum = (nums: number[]): number[] => {
  for (let i = 1; i < nums.length; i++) {
    nums[i] += nums[i - 1]
  }
  return nums
}

// console.log(runningSum([1, 2, 3, 4]))
// console.log(runningSum([1, 1, 1, 1, 1]))
// console.log(runningSum([3, 1, 2, 10, 1]))

File: src/Running_Sum_of_1d_Array/solution.py
class Solution:
    def runningSum(self, nums: list[int]) -> list[int]:
        running_sum = 0
        for i in range(len(nums)):
            running_sum += nums[i]
            nums[i] = running_sum
        return nums
    
# print(Solution().runningSum([1, 2, 3, 4]))
# print(Solution().runningSum([1, 1, 1, 1, 1]))
# print(Solution().runningSum([3, 1, 2, 10, 1]))
    